zipline/data/data_portal.py

```diff
@@ -338,6 +339,75 @@ def get_spot_value(self, asset, field, dt, data_frequency):
                 return self._get_minute_spot_value(
                     asset, column_to_use, dt)
 
+    def _get_adjusted_value(self, asset, field, dt,
+                            perspective_dt,
+                            data_frequency):
+        """
+        Private method that returns a scalar value representing the value
+        of the desired asset's field at either the given dt.
+
+        Parameters
+        ---------
+        asset : Asset
+            The asset whose data is desired.gith
+
+        field: string
+            The desired field of the asset.  Valid values are "open",
+            "open_price", "high", "low", "close", "close_price", "volume", and
+            "price".
+
+        dt: pd.Timestamp
+            The timestamp for the desired value.
+
+        perspective_dt : pd.Timestamp
+            The timestamp from which the data is being viewed back from.
+
+        data_frequency: string
+            The frequency of the data to query; i.e. whether the data is
+            'daily' or 'minute' bars
+
+        Returns
+        -------
+        The value of the desired field at the desired time.
+        """
+        if isinstance(asset, int):
+            asset = self._asset_finder.retrieve_asset(asset)
+
+        spot_value = self.get_spot_value(asset, field, dt, data_frequency)
+
+        if isinstance(asset, Equity):
+            adjs = []
+            split_adjustments = self._get_adjustment_list(
+                asset, self._splits_dict, "SPLITS"
+            )
+            for adj_dt, adj in split_adjustments:
+                if adj_dt < dt <= perspective_dt:
+                    if field != 'volume':
+                        adjs.append(adj)
+                    else:
+                        adjs.append(1.0 / adj)
+
+            if field != 'volume':
+                merger_adjustments = self._get_adjustment_list(
+                    asset, self._mergers_dict, "MERGERS"
+                )
+                for adj_dt, adj in merger_adjustments:
+                    if adj_dt < dt <= perspective_dt:
+                        adjs.append(adj)
+
+                div_adjustments = self._get_adjustment_list(
+                    asset, self._dividends_dict, "DIVIDENDS",
+                )
+                for adj_dt, adj in div_adjustments:
+                    if adj_dt < dt <= perspective_dt:
```

Comment:
since dividends (and all the other adjustments) come back in sorted order, is it worth breaking out of this list the first time you see an adj_dt that is greater than `dt` or `perspective_dt`?